coralproject/talk
# [CORL-2476]: default stories and dashboard to sites within scope

## What does this PR do?

These changes make it so that on the `Stories` and `Dashboard` tabs in the admin, site moderators only see stories and dashboard information for the site(s) that they moderate.

It also migrates all `stories.spec.tsx` tests over to RTL, including adding the rest of the previously commented-out test for the story info drawer.

## What changes to the GraphQL/Database Schema does this PR introduce?

In the stories connection query, it changes the `siteID` input to `siteIDs` to accept an array of site ids. This allows for both filtering stories by one siteID as it currently does, as well as multiple site IDs, as was needed to be able to filter stories for the `Stories` tab by site IDs if a moderator moderates more than one site, for example.

## Does this PR introduce any new environment variables or feature flags? 

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can go to the `Stories` tab as a site moderator. Test as both a single-site moderator and multi-site site moderator. Single-site moderators should only see stories for the 1 site that they moderate shown. Search by story title or author and see that the results shown are only for that story. See that the query string in the url still works as expected to search (ex. `http://localhost:8080/admin/stories?q=search`) the stories. See that the `Search by site name` option isn't shown for single-site moderators since they only see stories for 1 site anyway. Multi-site moderators should also only see stories for their sites, but they also see the `Search by site name` option.

See that admins and org moderators see stories for all sites and the `Search by site name` option on the `Stories` tab.

Also go to the `Dashboard` tab. See that single-site moderators see only the dashboard for the site they moderate and there is no dropdown carat shown to switch to another site beside the site name. See that multi-site moderators have only options for seeing dashboards for the sites they moderate. See that admin and org moderators can switch between all sites' dashboards. 
 
## How do we deploy this PR?
